app/code/Magento/Cms/Api/BlockManagementInterface.php

```diff
@@ -0,0 +1,23 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Cms\Api;
+
+/**
+ * CMS Block management interface
+ * @api
+ */
+interface BlockManagementInterface
+{
+    /**
+     * Load block data by given block identifier.
+     *
+     * @param string $identifier
+     * @param int|null $storeId
+     * @return \Magento\Cms\Api\Data\BlockInterface
+     */
+    public function getByIdentifier(string $identifier, $storeId = null) : \Magento\Cms\Api\Data\BlockInterface;
```

Comment:
hey @damiantomczak 
I have two points how we can improve current interfaces: 

1. **Naming**
Interfaces should have an explicit name which represents what particular entity responsible for. Reading the name of the interface you should have a pretty close guess what you will find among the methods which are exposed as a contract of this interface. 
Like, by the name of `RepositoryInterface` you understand what kind of methods would be inside.
Reading `ManagementInterface`  you don't have any clue what would be inside. Because `Manager` is the same **hack** as `Helper`. And Helpers are forbidden in Magento 2, because developers use helpers when they don't find a correct place to put their business logic, thus they end up creating Helper. The same story happens with Manager naming.

Probably if we didn't have the Repository here - we would put newly created methods to the Repository. But taking into account that Repositories already exist - we are not allowed to extend their contract because of Backward Compatibility perspectives.

Also, a problem could appear in future when someone decides to add another method, like `SaveByIdentity`. But how will he call this API? The name `Manager` - already occupied by current service. So, the one will need to find another name which is not reserved yet.  Any time when you hard to find a name for the entity you would like to introduce - it's a first **"smell"**, that something is going wrong.

What we can do here -  is well described in this [article](https://github.com/magento-engcom/msi/wiki/Service-Contracts#magento-repositories-basics).

You can introduce Command Interface like this one:
```
interface GetIdentity
{
   public function execute(string $id, $scope);
}
```
Each such interface describes a Command. It's pretty clear what it does. And we are getting closer to CQRS. Neither we violate Backward Compatibility adding such services.

2. Another point to improve is optional parameters. They are another thing which should be avoided. Because optional parameters lead to conditional logic in the implementation. 
Btw here is this logic in your case
```
       if ($storeId === null) {
           $storeId = $this->storeManager->getStore()->getId();
      }
```

Here we could possibly have two approaches:
- resolve a scope and explicily pass it to the method
- let the method to implicitly resolve scope for you.

The second option is error prone, because un evident and could lead to different bugs because of wrong scope selection.

Thus, it's recommended to follow 1st approach, and not allow the implementation to resolve scope (aka violate single responsibility principle) at the level of required inbound $scope parameter being passed.

Reply:
I'm very thankful for your deep explanation, of course I can not disagree with you so I prepared corresponding fixes to my pull request. Please check again and say me what you think.